### PR TITLE
315 configurable base url

### LINKED ISSUE #315: Configurable base url

### PR DESCRIPTION
**Pull request by paibamboo**
Closes #315

### CODE REVIEW

File: src/app/routes/routes.ts [hunk 1]
```diff
@@ -15,37 +15,36 @@ type RouteConfig = Record<RoutablePages, Omit<IRoute, "name">>;
...
   homePage: {path: "/"},
   starsPage: {path: "/stars"}
 };
 
-export const routes = getRoutes(config);
+let cachedRoutes: Record<RoutablePages, IRoute> = null;
+let cachedBaseUrl = "";
+
+export function getRoutes(baseUrl: string = ""): Record<RoutablePages, IRoute> {
+  if (!cachedRoutes || cachedBaseUrl !== baseUrl) {
```
Comment (cyberhck) [inline src/app/routes/routes.ts]: actually I'd have prefer not to cache here, but when we consume these, do a const `routes = getRoutes()` cacheBaseUrl is acting like a global (but not really, global to this file I mean)

File: src/app/routes/configureRouter.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import createRouter, {Router} from "router5";
 import browserPlugin from "router5-plugin-browser";
 import {MiddlewareFactory} from "router5/types/types/router";
-import {routes} from "./routes";
+import {getRoutes} from "./routes";
 
-export function configureRouter(): Router {
-  const router = createRouter(Object.keys(routes).map((key) => routes[key]));
+export function configureRouter(baseUrl: string = ""): Router {
+  const router = createRouter(Object.keys(getRoutes(baseUrl)).map((key) => getRoutes(baseUrl)[key]));
```
Comment (cyberhck) [inline src/app/routes/configureRouter.ts]: move getRoutes(baseUrl) to a constant and use that on both sides?

File: src/app/containers/Header.tsx [hunk 3]
```diff
@@ -45,22 +45,22 @@ class Header extends React.Component<IStateToProps> {
       <nav className={classNames.nav}>
         <ul>
           <li>
-            <ConnectedLink activeClassName={classNames.activeLink} routeName={routes.homePage.name}>
+            <ConnectedLink activeClassName={classNames.activeLink} routeName={getRoutes().homePage.name}>
```
Comment (cyberhck) [inline src/app/containers/Header.tsx]: can we have a `const routes = getRoutes()` instead?

**Comment by cyberhck** [comment]
apart from some minor comments, everything else looks good :)

**Comment by paibamboo** [comment]
@cyberhck Feedback updated. Please check again.

**Pull request merged**